ledger.go

```diff
@@ -200,29 +210,122 @@ func (l *Ledger) Addresses(hrp string, addresses int) ([]*Address, error) {
 // SignHash attempts to sign the [hash] with the provided path [addresses].
 // [addresses] are appened to the [pathPrefix] (m/44'/9000'/0'/0).
 func (l *Ledger) SignHash(hash []byte, addresses []uint32) ([][]byte, error) {
-	msgHash := []byte{
-		CLA,
-		INSSignHash,
-		0x0,
-		0x0,
-	}
 	pathBytes, err := bip32bytes(pathPrefix, 3)
 	if err != nil {
 		return nil, err
 	}
 	data := []byte{byte(len(addresses))}
 	data = append(data, hash...)
 	data = append(data, pathBytes...)
-	msgHash = append(msgHash, byte(len(data)))
-	msgHash = append(msgHash, data...)
+	data = append([]byte{byte(len(data))}, data...)
 	fmt.Printf("signing hash: %X\n", hash)
-	resp, err := l.device.Exchange(msgHash)
+	resp, err := l.SendToLedger(CLA, INSSignHash, 0x00, 0x00, data)
 	if err != nil {
 		return nil, err
 	}
 	if !bytes.Equal(resp, hash) {
 		return nil, fmt.Errorf("returned hash %x does not match requested %x", resp, hash)
 	}
 
-	return l.collectSignatures(addresses)
+	return l.collectSignatures(addresses, INSSignHash, 0x01, 0x81)
+}
+
+func (l *Ledger) PrepareChunks(txn []byte, addresses []uint32, changePath []uint32) ([][]byte, error) {
+	if txn == nil || addresses == nil {
```

Comment:
Is there a reason we want to treat:
`len(txn) == 0` as different than `txn == nil`? Same for `addresses`.
I've always avoided treating the `nil` slice as a special flag... because it ends up being error prone.